test/NuGet.Core.Tests/NuGet.DependencyResolver.Core.Tests/RemoteWalkContextTests.cs

```diff
@@ -44,19 +44,27 @@ public void FilterDependencyProvidersForLibrary_WhenPackageSourceMappingIsNotEna
             Assert.Equal(context.RemoteLibraryProviders, providers);
         }
 
-        [Fact]
-        public void FilterDependencyProvidersForLibrary_WhenPackageSourceMappingIsEnabledReturnsOnlyApplicableProviders_Success()
+        [Theory]
+        [InlineData("Source1", "Source1")]
+        [InlineData("SOURCE1", "SOURCE1")]
+        [InlineData("Source1", "source1")]
+        [InlineData("source1", "Source1")]
+        [InlineData("SOURCE1", "Source1")]
+        [InlineData("Source1", "SOURCE1")]
+        public void FilterDependencyProvidersForLibrary_WithDifferentCasing_WhenPackageSourceMappingIsEnabledReturnsOnlyApplicableProviders_Success(string packageSourceMappingName, string providerName)
         {
             //package source mapping configuration
-            Dictionary<string, IReadOnlyList<string>> patterns = new();
-            patterns.Add("Source1", new List<string>() { "x" });
-            patterns.Add("Source2", new List<string>() { "y" });
+            Dictionary<string, IReadOnlyList<string>> patterns = new()
+            {
+                { packageSourceMappingName, new List<string>() { "x" } },
+                { "Source2", new List<string>() { "y" } }
+            };
             PackageSourceMapping sourceMappingConfiguration = new(patterns);
 
             var context = new TestRemoteWalkContext(sourceMappingConfiguration, NullLogger.Instance);
 
             // Source1
-            var remoteProvider1 = CreateRemoteDependencyProvider("Source1");
+            var remoteProvider1 = CreateRemoteDependencyProvider(providerName);
```

Comment:
nit: an explicit type would be useful as well

```suggestion
            var remoteProvider = CreateRemoteDependencyProvider(providerName);
```